src/app/task.service.ts

```diff
@@ -0,0 +1,40 @@
+import {Injectable} from '@angular/core';
+import {HttpClient} from "@angular/common/http";
+import {Task} from "./task";
+import {BehaviorSubject} from "rxjs";
+
+@Injectable({
+  providedIn: 'root'
+})
+export class TaskService {
+  private taskSubject = new BehaviorSubject<Task[]>([]);
+  tasks$ = this.taskSubject.asObservable();
+  FILE_PATH: string = '../../assets/dummy-tasks.json';
+
+  constructor(private http: HttpClient) {
+    this.loadInitialTasks();
+  }
+
+  loadInitialTasks(): void {
+    this.http.get<{ data: { tasks: Task[] } }>(this.FILE_PATH)
+      .subscribe(response => {
```

Comment:
avoid subscribing in services as it's much harder to maintan subscriptions here, instead you can subscribe and unsubscribe on the component level